### PR TITLE
Dev 18 move book state

### PR DESCRIPTION
**Pull request by Four58**

### CODE REVIEW

**Review by Four58 — COMMENTED** [review]
good job myself

File: src/components/Book/Book.tsx [hunk 1]
```diff
@@ -13,6 +16,7 @@ type props = {
   like: number;
   isOwner: boolean;
   coverImage: string | null;
+  status: string;
```
Comment (swsd2544) [inline src/components/Book/Book.tsx:19]: Change type to BookStatus

File: src/pages/[penname]/book/[bookId]/index.tsx [hunk 2]
```diff
@@ -71,6 +71,7 @@ type props = InferGetServerSidePropsType<typeof getServerSideProps>;
 
 const BookContent = ({ bookId }: props) => {
   const router = useRouter();
+  const penname = router.query.penname;
```
Comment (swsd2544) [inline src/pages/[penname]/book/[bookId]/index.tsx:74]: Get penname from getServerSideProps instead (the same as bookid)

**Review by swsd2544 — CHANGES_REQUESTED** [review]

File: src/pages/[penname]/book/[bookId]/status.tsx [hunk 3]
```diff
@@ -0,0 +1,674 @@
...
+  const {
+    imageData: bookWallpaper,
+    uploadHandler: setBookWallpaper,
+    resetImageData: resetBookWallpaper,
+  } = useImageUpload();
+  const router = useRouter();
+  const penname = router.query.penname;
+  const utils = api.useContext();
+  const { data: book } = api.book.getData.useQuery({ id: bookId });
+  const { data: categories } = api.category.getAll.useQuery();
```
Comment (swsd2544) [inline src/pages/[penname]/book/[bookId]/status.tsx:79]: prefetch categories in getServerSideProps

File: src/pages/[penname]/book/[bookId]/status.tsx [hunk 4]
```diff
@@ -0,0 +1,674 @@
...
+  const [addedCategories, setAddedCategories] = useState(
+    book?.categories.map((data) => data.category) || []
+  );
+  const deleteBook = api.book.delete.useMutation({
+    onSuccess: () => {
+      void utils.book.invalidate();
+    },
+  });
+  const { data: collaborators } = api.user.getBookCollaborators.useQuery({
```
Comment (swsd2544) [inline src/pages/[penname]/book/[bookId]/status.tsx:88]: prefetch this in getServerSideProps

**Review by swsd2544 — CHANGES_REQUESTED** [review]

**Comment by swsd2544** [comment]
Books in [penname]/books/index.tsx is paginated, please add the pagination feature

File: src/pages/[penname]/book/[bookId]/status.tsx [hunk 5]
```diff
@@ -0,0 +1,674 @@
...
+    imageData: bookWallpaper,
+    uploadHandler: setBookWallpaper,
+    resetImageData: resetBookWallpaper,
+  } = useImageUpload();
+  const router = useRouter();
+  const penname = router.query.penname;
+  const utils = api.useContext();
+  const { data: book } = api.book.getData.useQuery({ id: bookId });
+  const { data: categories } = api.category.getAll.useQuery();
+  const [addedCategories, setAddedCategories] = useState(
```
[the remark this answers is not in the document]
Comment (Four58) [reply src/pages/[penname]/book/[bookId]/status.tsx:80]: no

File: src/pages/[penname]/book/[bookId]/status.tsx (hunk 4 above)
Comment (Four58) [reply src/pages/[penname]/book/[bookId]/status.tsx:88]: what is this

File: src/pages/[penname]/book/[bookId]/status.tsx (hunk 3 above)
Comment (Four58) [reply src/pages/[penname]/book/[bookId]/status.tsx:79]: okay

File: src/pages/[penname]/book/[bookId]/index.tsx (hunk 2 above)
Comment (Four58) [reply src/pages/[penname]/book/[bookId]/index.tsx:74]: i'll try

File: src/components/Book/Book.tsx (hunk 1 above)
Comment (Four58) [reply src/components/Book/Book.tsx:19]: ok

File: src/pages/[penname]/book/[bookId]/status.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (swsd2544) [reply src/pages/[penname]/book/[bookId]/status.tsx:80]: fix this first

File: src/pages/[penname]/book/[bookId]/status.tsx (hunk 4 above)
Comment (swsd2544) [reply src/pages/[penname]/book/[bookId]/status.tsx:88]: references:
- https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props
- https://trpc.io/docs/nextjs/ssg-helpers

**Review by swsd2544 — APPROVED** [review]
LGTM!

**Pull request merged**